src/tags/map.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @Author: RagingLink 
+ * @Date: 2020-06-03 23:26:05
+ * @Last Modified by: RagingLink
+ * @Last Modified time: 2020-07-17 22:17:35
+ *
+ * This project uses the AGPLv3 license. Please read the license file before using/adapting any of the code.
+ */
+
+const Builder = require('../structures/TagBuilder');
+
+module.exports =
+    Builder.ArrayTag('map')
+        .withArgs(a => [
+            a.require('variable'),
+            a.require('array'),
+            a.require('function')
+        ])
+        .withDesc('Provides a way to populate an array by executing a function on each of its elements,' +
+            ' more info [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)\n' +
+            'For every element in `array`, `variable` will be set and `function` will be executed. The output of `function`' +
+            ' will be the new value of the element. If `array` is a variable it will update the variable. Otherwise,' +
+            ' it will simply output the new array.'
+        )
+        .withExample(
+            '{map;~item;["apples","oranges","pears"];{upper;{get;~item}}}',
+            '["APPLES","ORANGES","PEARS"]'
+        )
+        .resolveArgs(0, 1)
+        .whenArgs('0-2', Builder.errors.notEnoughArguments)
+        .whenArgs('3', async function (subtag, context, args) {
+            let varName = args[0],
+                arr = await bu.getArray(context, args[1]),
+                array = Array.from(arr.v),
+                newArray = [];
+            
+            let remaining = context.state.limits.map || {loops : NaN};
+            for (const item of array) {
+                remaining.loops--;
+                if (!(remaining.loops >= 0)) {
```

Comment:
reverse the condition : `remaining.loops < 0`

Reply:
```// (remaining.loops < 0) would not work due to the comparison behaviours of NaN``` - Comment from `{repeat}`
If I'd remove the `{loops : NaN}`, changing it to `remaining.loops < 0` would be a lot cleaner. But as I mentioned in my previous reply, I'm not even sure if I should remove it.